client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -212,66 +222,50 @@ private Response performRequest(final NodeTuple<Iterator<Node>> nodeTuple,
         } catch(Exception e) {
             RequestLogger.logFailedRequest(logger, request.httpRequest, context.node, e);
             onFailure(context.node);
-            Exception cause = unwrapExecutionException(e);
+            Exception cause = extractAndWrapCause(e);
             addSuppressedException(previousException, cause);
             if (nodeTuple.nodes.hasNext()) {
                 return performRequest(nodeTuple, request, cause);
             }
             if (cause instanceof IOException) {
```

Comment:
Huh. I think you might be able to drop these `instanceof` if you moved some stuff into `extractAndWrapCause`. You already know the type. I'm not sure you *need* to do that though. It'd be slightly cleaner, I think, but it isn't worth holding up the PR for it.

Reply:
the thing is that there are cases where I don't re-throw the exception gotten from extractAndWrapCause, I may pass it over to performRequest is I can retry on anothe rnode :)

Reply:
Ah ha! I see that two lines up. That makes sense.